assets/src/dashboard/app/views/storyAnimTool/index.js

```diff
@@ -161,27 +162,32 @@ function StoryAnimTool() {
 
   const handleAddOrUpdateAnimation = useCallback(
     (animation) => {
-      const story = { ...activeStory };
-      const animationWithTargets = {
-        ...animation,
-        targets: [...Object.values(selectedElementIds)],
-      };
-
-      const animations = story.pages[activePageIndex].animations || [];
-      const index = animations.findIndex(
-        (a) => a.id === animationWithTargets.id
+      const story = Object.values(selectedElementIds).reduce(
+        (story, selectedElementId, i) => {
+          const animationWithTarget = {
+            ...animation,
+            targets: [selectedElementId],
+            id: i === 0 ? animation.id : uuidv4(),
+          };
+
+          const animations = story.pages[activePageIndex].animations || [];
+          const index = animations.findIndex(
+            (a) => a.id === animationWithTarget.id
+          );
+
+          if (index < 0) {
+            // not found, push it to the end
+            animations.push(animationWithTarget);
+          } else {
+            // found, overwrite at index
+            animations[index] = animationWithTarget;
+          }
+
+          story.pages[activePageIndex].animations = [...animations];
```

Comment:
Do we want a deep copy here or just a new array with elements?

Reply:
I honestly don't even see the need for all this copying back n forth and think it's fine to operate on the accumulator directly. I was just trying to change as little as possible and this was in the old code

Happy to refactor & clean up a bit

Also mind clarifying on deep copy vs new array with elements? don't think I'm 100% following you there

Reply:
I don't _think_ we need a deep copy here, and should be more performant without one